### PR TITLE
[Improvement](join) some improvement of hash join

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
图片

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run ckbench

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run p0

File: be/src/vec/exec/join/process_hash_table_probe_impl.h [hunk 1]
```diff
@@ -277,6 +277,21 @@ Status ProcessHashTableProbe<JoinOpType>::do_process(HashTableType& hash_table_c
             }
         }
 
+        _build_side_hash_values.resize(probe_rows);
```
Comment (HappenLee) [inline be/src/vec/exec/join/process_hash_table_probe_impl.h:280]: why the name is _build_side_hash_values? the probe side

File: be/src/vec/common/hash_table/ph_hash_map.h [hunk 2]
```diff
@@ -178,14 +178,18 @@ class PHHashMap : private boost::noncopyable {
     }
 
     template <typename KeyHolder>
-    void ALWAYS_INLINE emplace(KeyHolder&& key_holder, LookupResult& it, size_t hash_value,
-                               bool& inserted) {
+    void ALWAYS_INLINE emplace(KeyHolder&& key_holder, LookupResult& it, bool& inserted,
```
Comment (HappenLee) [inline be/src/vec/common/hash_table/ph_hash_map.h:181]: why change the order?

File: be/src/vec/exec/join/process_hash_table_probe_impl.h [hunk 3]
```diff
@@ -305,18 +320,23 @@ Status ProcessHashTableProbe<JoinOpType>::do_process(HashTableType& hash_table_c
                     }
                 }
```
Comment (HappenLee) [inline be/src/vec/exec/join/process_hash_table_probe_impl.h:321]: maybe need to do the same opt in do_process_with_join_conjunct

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]

**Comment by BiteTheDDDDt** [comment]
Do not merge this pr to 2.0

**Pull request merged**